src/library/scala/concurrent/BatchingExecutor.scala

```diff
@@ -255,6 +255,12 @@ private[concurrent] trait BatchingExecutor extends Executor {
       if (i < BatchingExecutorStatics.syncPreBatchDepth) {
         tl.set(java.lang.Integer.valueOf(i + 1))
         try submitForExecution(runnable) // User code so needs to be try-finally guarded here
+        catch {
```

Comment:
for reference, this makes it consistent with https://github.com/scala/scala/pull/7783/files#diff-4a66ae20fe1614a8edbd797892372d9fL211